using-timescaledb/adaptive-chunking.md

```diff
@@ -0,0 +1,123 @@
+data is its automatic partitioning scheme that splits the data across
+a number of N-dimensional *chunks*. However, the performance will only
+be good if the system achieves the "right" *chunk size* that won't
+blow up the memory budget. Oversized chunks result in slow INSERT
+performance when updating indexes and also negatively affect query
+speed.
+
+Unfortunately, the chunk size cannot be set directly as the size is
+governed by the constraints on the chunk and the data
+rate. Dynamically altering constraints on an existing chunk to achieve
+a certain chunk size is an expensive and often impossible operation,
+especially in face of changing data rates. Thus, a change in data rate
+might either "overfill" or "underfill" a chunk.
+
+Thus, setting a good `chunk_time_interval` is crucial to achieving a
+good chunk size, since it determines the time constraint on
+chunks. Fortunately, the `chunk_time_interval` can also be updated on
+an existing hypertable, although this new setting only applies to
+chunks created after the change. A conflict resolution mechanism
+within TimescaleDB automatically detects any "collisions" between new
+and old chunks due to the change in partitioning, and cuts chunks to
+fit within existing constraints.
+
+*Adaptive chunking* leverages the ability to set new
+`chunk_time_interval`s, automating this process.  The system picks the
+first *open* (time) dimension to adapt, which is the same dimension
+that `chunk_time_interval` works on.  With adaptive chunking, however,
+the user instead sets a `chunk_target_size`, e.g., `1GB` and the
+system will try to adapt the `chunk_time_interval` on new chunks to
+reach this target chunk size using an algorithm.
+
+>:TIP: Setting a reasonable initial `chunk_time_interval` is still
+important because it allows the adaptive algorithm to quicker reach
+the target chunk size. Setting a too large `chunk_time_interval` is
+worse than setting a too small one, because the system will be
+committed to that larger interval until the next chunk. If no
+`chunk_time_interval` is set with adaptive chunking, the default is 1
+day.
+
+The default algorithm to calculate the next chunk's interval takes
+into account the *fill factor* of previous chunks, i.e., if they were
+over-filled or under-filled, and changes the interval accordingly. The
+system picks the first *open* (time) dimension to adapt. A user can
+set its own function to calculate the next `chunk_time_interval`,
+allowing customized algorithms.
+
+>:TIP: It is recommended that hypertables with adaptive chunking
+enabled has an index on the dimension being adapted, typically the
+"time" dimension. Such an index is created by default, unless
+otherwise specified. The index is used by the default adaptive
+chunking algorithm to get the `min` and `max` value from a chunk to
+calculate its fill factor. Without an index, it has to resort to a
+heap scan, which is expensive. The system will print `WARNING`s in the
```

Comment:
`to a heap scan (i.e., scanning all the tuples in the chunk)` for readers who may not be familiar with database internals